src/compiler/checker.ts

```diff
@@ -1474,6 +1475,11 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
     var checkBinaryExpression = createCheckBinaryExpression();
     var emitResolver = createResolver();
     var nodeBuilder = createNodeBuilder();
+    var evaluate = createEvaluator({
+        evaluateElementAccessExpression,
+        evaluateEntityNameExpression,
+        onNumericLiteral: checkGrammarNumericLiteral,

```

Comment:
It feels strange to have this callback. This only needs the AST node and nothing else. Can we just walk the nodes like other grammar checks?

Reply:
I was trying to introduce as little extra overhead as possible, I agree that callback is kinds of the odd one out, but without an extra walk I'm not sure how we could ensure that check is performed in the same way. 

The check does appear to happen in other places though. Enum members should go through `checkEnumMember` which will call `checkGrammarNumericLiteral` during expression checking. Similarly for template literals. So the check might be redundant for a full check. I'll test and see.

Reply:
Just tested .. removing the check from here does not fail any test 😕. I do think it ends up being called through other paths.